src/argus_htmx/incidents/customization.py

```diff
@@ -0,0 +1,61 @@
+"""
+Definitions and defaults for UI customization on the incidents page.
+
+Currently customizable UI elements:
+- table columns: hide default columns and/or show additional columns
+
+Expected to be customizable in the future [WIP list]:
+- table row colors
+- table row effects
+- add custom components to the incidents page
+"""
+
+from dataclasses import dataclass
+from typing import Optional, OrderedDict
+
+
+@dataclass
+class IncidentField:
+    """Class for keeping track of columns in the incidents table."""
+
+    name: str  # identifier
+    label: str  # display value
+    cell_template: str
+    context: Optional[dict] = None
+
+
+DEFAULT_FIELDS = OrderedDict[str, IncidentField]([
+    ("row_select", IncidentField("checkbox", "Selected", "htmx/incidents/_incident_checkbox.html")),
+    ("start_time", IncidentField("start_time", "Timestamp", "htmx/incidents/_incident_start_time.html")),
+    ("status", IncidentField("status", "Status", "htmx/incidents/_incident_status.html")),
+    ("level", IncidentField("level", "Severity level", "htmx/incidents/_incident_level.html")),
+    ("source", IncidentField("source", "Source", "htmx/incidents/_incident_source.html")),
+    ("description", IncidentField("description", "Description", "htmx/incidents/_incident_description.html")),
+    ("links", IncidentField("links", "Actions", None))
+])
+
+# TODO remove before alpha release
+# Fields for demo/debug/dev purposes
+TEMP_FIELDS = {
+    "id": IncidentField("id", "ID", "htmx/incidents/_incident_pk.html"),
+    "tag": IncidentField("tag", "Tag", "htmx/incidents/_incident_tag.html", context={"tag": "location"}),
+    "ack": IncidentField("ack", "Ack", "htmx/incidents/_incident_ack.html")
+}
```

Comment:
These are all useful even for us. Rename to OTHER_FIELDS or EXTRA_FIELDS instead?

Reply:
Looking closer, the "tag" IncidentField needs to have customizable key.

We have use of an ack-column, geant has use of a grouped ack-column.

Reply:
The problem is the lookup needed. This lookup: `incident.acks.active().filter(group__name=GROUP)` cannot/should not be done in a template.

The tags lookup already works.